### ISSUE TITLE
file module does not set the right owner when creating nested directories

### ISSUE THREAD

**Issue by universal** [ansible/ansible#3300]
Hello,

When run with `sudo: yes` and creating a nested directory structure: The file module only sets correct owner/group for the last directory in the chain. An example of this can be seen in the simple playbook:

``` yaml

---
- hosts: some_host
  user: some_user
  sudo: yes
  tasks:
    - name: ensure Backup dir is present
      file: state=directory path=/home/git/Backup/models owner=git group=git recurse=yes
```

The /home/git/Backup/models directory is owned by git:git, /home/git/Backup is owned by root:root, where i would expect it to be also owned by git:git.

This happens on a current checkout from the devel branch. If you need any further information, i'm happy to supply whatever is needed.

**Comment by mpdehaan** [comment]
This is true because we don't really know WHAT to set the owner for.

For instance, suppose in one case / exists and you ask to create /asdf/bar/baz/glorp.    The permissions of glorp are not neccessarily those of / or /asdf

Should /asdf exist, the results are also different.

Thus, in order to make sure you are correctly modelling the desired end state, it is neccessary to specify the owner for everything along the chain.

Basically it's the problem of the file module _needing_ to be idempotent, which means it gets you to the same ending state every single time.

So what I would do here is have the file module call here, but also set the owner of /home/git/Backup explicitly as well.

If you have any further questions on this, I would like to discuss further, can you stop by the mailing list?

Thanks!
